cli/internal/env/env.go

```diff
@@ -30,33 +30,36 @@ func getEnvPairsFromKeys(envKeys []string, allEnvVars map[string]string) []strin
 	return hashableConfigEnvPairs
 }
 
-// getFrameworkEnvPairs returns a slice of all key=value pairs that match the given prefix
-func getEnvPairsFromPrefix(prefix string, allEnvVars map[string]string) []string {
-	hashableFrameworkEnvPairs := []string{}
+// getEnvPairsFromPrefix returns a slice of all key=value pairs that match the given prefix
+func getEnvPairsFromPrefix(includePrefix string, excludePrefix string, allEnvVars map[string]string) []string {
+	allEnvPairs := []string{}
 	for k, v := range allEnvVars {
-		if strings.HasPrefix(k, prefix) {
-			hashableFrameworkEnvPairs = append(hashableFrameworkEnvPairs, fmt.Sprintf("%v=%v", k, v))
+		if excludePrefix != "" && strings.HasPrefix(k, excludePrefix) {
+			continue
+		}
+		if strings.HasPrefix(k, includePrefix) {
+			allEnvPairs = append(allEnvPairs, fmt.Sprintf("%v=%v", k, v))
 		}
 	}
-	return hashableFrameworkEnvPairs
+	return allEnvPairs
 }
 
-// getEnvPairsFromPrefixes returns a slice containing key=value pairs for all frameworks
-func getEnvPairsFromPrefixes(prefixes []string, allEnvVars map[string]string) []string {
-	allHashableFrameworkEnvPairs := []string{}
-	for _, frameworkEnvPrefix := range prefixes {
-		hashableFrameworkEnvPairs := getEnvPairsFromPrefix(frameworkEnvPrefix, allEnvVars)
-		allHashableFrameworkEnvPairs = append(allHashableFrameworkEnvPairs, hashableFrameworkEnvPairs...)
-
+// getEnvPairsFromPrefixes returns a slice containing key=value pairs
+func getEnvPairsFromPrefixes(includePrefixes []string, excludePrefix string, allEnvVars map[string]string) []string {
+	allEnvPairs := []string{}
+	for _, includePrefix := range includePrefixes {
+		hashableFrameworkEnvPairs := getEnvPairsFromPrefix(includePrefix, excludePrefix, allEnvVars)
```

Comment:
I would probably have inlined this logic, because I get confused when there are two functions named the same, but just a style preference (and I see that there were already two functions)

Reply:
This is definitely a path-dependence outcome. It'd have been fine the other way but the path we took to get here had this outcome.